dali/kernels/reduce/reduce_gpu_impl.cuh

```diff
@@ -44,6 +44,27 @@ namespace kernels {
 /// @brief Implementation details of reduction kernels
 namespace reduce_impl {
 
+template <typename Iterator, typename Predicate>
```

Comment:
some doxygen to multi_partition would be nice

Reply:
Actually, I could even move it to some utils.

Reply:
Done.